src/extension.in.h

```diff
@@ -30,10 +30,37 @@ typedef struct
 /*	configure extension version in makefile */
 #define EXTVERSION "@EXTVERSION@"
 
-/* base type in postgres is Datum, and we cannot fit 8 bytes on all platforms
-   so we use pointers */
-#define PG_RETURN_H3_INDEX_P(x) PG_RETURN_POINTER(x)
-#define PG_GETARG_H3_INDEX_P(x) (H3Index *) PG_GETARG_POINTER(x)
+/*
+ * DatumGetH3Index
+ *		Returns H3 index value of a datum.
+ *
+ * Note: this macro hides whether h3 index is pass by value or by reference.
+ */
+
+#ifdef USE_FLOAT8_BYVAL
```

Comment:
I haven't checked, but if there is something like USE_INT8_BYVAL it might be irrelevantly more beautiful here.

Reply:
I thought the same, but as far as I can tell `USE_FLOAT8_BYVAL` is the only flag used (including for int64).